doc/source/library_description/packaging.rst

```diff
@@ -2,73 +2,230 @@
+
+
+Python scripts, modules, sub-packages and packages
+--------------------------------------------------
+
+It is important to understand the difference between Python scripts, modules,
+sub-packages and packages:
+
+* ``Script``: any Python file holding logic source code.
+* ``Module``: any Python script hosted next to an ``__init__.py`` file.
+* ``Sub-package``: any directory holding various Python modules.
+* ``Package``: any directory which contains Python modules and sub-packages.
+
+The following structure is shown to better explain previous concepts:
+
+.. code:: bash
+
+    .
+    ├── src
+    │   └── package
+    │       ├── subpackage_a
+    │       │   ├── __init__.py
+    │       │   └── module_c.py
+    │       ├── __init__.py
+    │       ├── module_a.py
+    │       └── module_b.py
+    ├── LICENSE
+    ├── README.rst
+    └── pyproject.toml
+
 
 Namespace Packaging
 -------------------
-A PyAnsys library uses `namespace packaging`_.
-Namespace packages allow a user to easily split subpackages from a package into
-a single, independent distribution.
+A PyAnsys library uses `namespace packaging`_.  Namespace packages allow a user
+to easily split sub-packages from a package into single, independent
+distributions.
+
+There are different approaches available for creating a namespace package. For
+the ``ansys`` namespace, we use the `PEP 420`_ `native namespace packages`_
+approach.
+
+Following previous namespace, the source directory of any `PyAnsys library`
+should look like this:
+
+.. code:: bash
 
-Three different approaches are currently available for creating a namespace package:
+    .
+    └── src
+        └── ansys
+            └── product
+                └── library
+                    └── __init__.py
 
-* `native namespace packages`_
-* pkgutil-style namespace packages
-* pkg_resources-style namespace packages
 
 Required Files
 --------------
 
-* README.rst file: Describes the purpose of the package.
+* ``README.rst`` file: Describes the purpose of the package.
   *The format of this file must be reStructuredText.*
 
-* LICENSE file: Specifies copyrights and required authorization.
+* ``LICENSE`` file: Specifies copyrights and required authorization.
 
-* setup.py file: Provides package information.
-  The presence of this file indicate that the package was likely created using ``disutils``,
-  which is the Python standard for building and distributing a Python package.
+* ``pyproject.toml`` file: Provides package information.
+  This file provides the package metadata, and defines how it is built.
+  There are different build backends available, such as `setuptools`_,
+  `poetry`_ and `flit`_.
 
+* ``src/ansys/product/library/__init__.py`` file: This file usually holds the
+  version of the package in a variable named ``__version__``. The value of this
+  variable can be parsed from the `pyproject.toml` file, so it is only specified
+  in one location.
```

Comment:
```suggestion
* ``src/ansys/product/library/__init__.py`` file: Usually holds the
  version of the package in a variable named ``__version__``. The value of this
  variable can be parsed from the ``pyproject.toml`` file so that it is only specified
  in one location.
```